libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivityDelegate.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2016-present, salesforce.com, inc.
+ * All rights reserved.
+ * Redistribution and use of this software in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * - Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.reactnative.ui;
+import com.facebook.react.ReactActivityDelegate;
+
+import javax.annotation.Nullable;
+
+
+public class SalesforceReactActivityDelegate extends ReactActivityDelegate {
+
+    private String appComponentName;
+    private SalesforceReactActivity salesforceReactActivity;
+    private boolean loaded = false;
+
+
+    public SalesforceReactActivityDelegate(SalesforceReactActivity activity, @Nullable String mainComponentName) {
+        super(activity, mainComponentName);
+        appComponentName = mainComponentName;
+        salesforceReactActivity = activity;
+    }
+
+
+    @Override
+    protected void loadApp(String appKey) {
+        loadReactAppOnceIfReady(appKey);
+    }
+
+    @Override
+    protected void onResume() {
+        loadReactAppOnceIfReady(appComponentName);
```

Comment:
Instead of overriding `onResume` here, maybe you could call `loadReactAppOnceReady` from `onResume` of `SalesforceReactActivity` once `client` is set if authentication is required (after the `peekRestClient` or `getRestClient` call). I think that should work for first time load too.